ipa-core/src/ff/boolean_array.rs

```diff
@@ -799,6 +820,41 @@ macro_rules! boolean_array_impl {
     };
 }
 
+macro_rules! boolean_array_impl_large {
+    ($modname:ident, $name:ident, $bits:tt, $deser_type:tt, $arraylength:ty) => {
+        boolean_array_impl!($modname, $name, $bits, $deser_type);
+
+        // used to convert into Fp25519
+        impl From<(u128, u128)> for $name {
```

Comment:
I don't think you can define this conversion for any boolean array. While it works for <32 byte arrays, it will panic at runtime.

Andy added a bunch of `const_assert` helpers and I think you can use them here to reject invocations for arrays > 32 bytes in size